src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -445,10 +445,17 @@ event::ptr primitive_inst::realloc_if_needed() {
     if (_node->is_type<input_layout>())
         return ev;
 
+    // read_value/assign nodes are supposed to always use variable memory
     if (auto stateful_prim = dynamic_cast<memory_state::variable*>(this)) {
         std::string variable_id = stateful_prim->variable_id();
         auto variable = get_network().get_variable(variable_id);
         variable.set_layout(actual_layout);
+        GPU_DEBUG_TRACE_DETAIL << id() << ": use variable memory " << variable.get_memory()
+                               << " (size=" << variable.get_memory()->size() << ")" << std::endl;
+        // For nodes that can be optimized, variable memory is used as output memory
+        // so there is no need for output memory reallocation
+        if (can_be_optimized())
+            return ev;
```

Comment:
hmm  BTW I think, if can_be_optimized(), not only state operators or concats, but also any other pprimitive should skip reorder.

Reply:
As you commented, in the case of reallocation skip for concat or other primitives in addition to state operators, for example, it should be updated considering existing condition that checks can_be_optimized() of concat for runtime concat optimization. So this can be applied in the next PR by considering various cases?